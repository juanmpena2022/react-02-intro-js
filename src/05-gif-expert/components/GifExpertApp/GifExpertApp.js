import { useState } from "react";
import { AddCategory } from "../AddCategory/AddCategory";
import { GifGrid } from "../GifGrid/GifGrid";

export const GifExpertApp = ({ defaultCategorias = ["One Punch"] }) => {
  const [categorias, setCategorias] = useState(defaultCategorias);

  //#region Funciones

  //#endregion

  return (
    <>
      <h2>GifExpertApp</h2>
      <AddCategory setCategorias={setCategorias} />
      <hr />
      <ol>
        {categorias.map((cat) => (
          <GifGrid key={cat} category={cat} />
        ))}
      </ol>
    </>
  );
};
