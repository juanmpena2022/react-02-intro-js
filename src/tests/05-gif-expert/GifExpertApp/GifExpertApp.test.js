import { shallow } from "enzyme";
import "@testing-library/jest-dom";
import { GifExpertApp } from "../../../05-gif-expert/components/GifExpertApp/GifExpertApp";

describe("pruebas del componente <GifExpertApp />", () => {
  test("snapshot", () => {
    const wrapper = shallow(<GifExpertApp />);
    expect(wrapper).toMatchSnapshot();
  });

  test("debe mostrar una lista de categorias", () => {
    const categorias = ["Dragon", "One Punch"];
    const wrapper = shallow(<GifExpertApp defaultCategorias={categorias} />);
    const gifGrids = wrapper.find('GifGrid');

    expect(gifGrids.length).toBe(categorias.length);
  });
});
