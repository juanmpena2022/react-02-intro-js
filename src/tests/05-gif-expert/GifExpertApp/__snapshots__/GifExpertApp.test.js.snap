// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`pruebas del componente <GifExpertApp /> snapshot 1`] = `
<Fragment>
  <h2>
    GifExpertApp
  </h2>
  <AddCategory
    setCategorias={[Function]}
  />
  <hr />
  <ol>
    <GifGrid
      category="One Punch"
      key="One Punch"
    />
  </ol>
</Fragment>
`;
